feat(esl-carousel): update autoplay plugin API to support pause behavior and global events observation (#3829)
exadel-inc/esl: packages/esl/src/esl-carousel/plugin/autoplay/README.md

 - `containerCls` (optional) – CSS class applied to the carousel container while autoplay is enabled.
 - `blockerSelector` (optional) – selector (ESLTraversingQuery) for items that, when activated, stop carousel autoplay. Defaults to `::find(esl-share[active], esl-note[active])`.
 - `watchEvents` (optional) – space-separated list of event names that toggle blocking state on the carousel when fired. Defaults to `esl:change:active`.
  - `blockBehaviour` (optional, default: `stop`) – how runtime blockers behave:
    * `stop` – clear the current cycle and start a new one after unblocking
    * `pause` – pause and resume the current cycle preserving the remaining time

### Public properties / state

 - `enabled` (boolean, read/write) – autoplay is enabled and not manually stopped.
 - `duration` (number, readonly) – parsed global duration (ms). Negative / NaN means disabled.
 - `effectiveDuration` (number, readonly) – current slide duration (per‑slide override or global). `<= 0` pauses only the current slide.
 - `remaining` (number, readonly) – remaining time of the current cycle (ms).

>  - `paused` (boolean, readonly) – autoplay is paused explicitly by user action and may be resumed.

Reviewer: ```suggestion
 - `paused` (boolean, readonly) – autoplay is paused explicitly by user action and can be resumed.
```

>  - `blocked` (boolean, readonly) – autoplay is currently blocked by viewport / interaction / blockers.

Reviewer: ```suggestion
 - `blocked` (boolean, readonly) – autoplay is temporarily suspended by external conditions (e.g. hidden viewport, user interaction, blockers).
```

 - `user:start:call`
 - `user:pause:call`
 - `user:stop:call`
 - `user:start:control`
 - `user:pause:control`
 - `user:stop:control`
 - `system:start:auto`
 - `system:pause:block`
 - `system:stop:block`
 - `system:stop:config`
 - `system:stop:slide-change`
 - `system:idle`

Notes:
 - `enabled = true` and `active = false` is normal when: base duration is `0`; current slide override is `0`/non‑positive; carousel not sufficiently visible; user is hovering or focused; navigation command currently not possible.
 - Base `duration = 0` does NOT disable the plugin – it just suppresses default cycles unless a per‑slide timeout overrides it.
 - Negative / invalid duration disables the plugin until changed.

### Manual control
Programmatic examples:
 - `carousel.autoplay.enabled = false` / `true`
 - `carousel.autoplay.stop()` / `start()`
 - `carousel.autoplay.pause()` to preserve remaining time

### Migration note for 6.2

In `6.2` autoplay controls are no longer configured through the `esl-carousel-autoplay` config object.

If you used:

```html
<esl-carousel esl-carousel-autoplay="{control: '.btn', controlBehaviour: 'pause'}"></esl-carousel>
```

> switch to a dedicated control mixin hosted on the control element:

Reviewer: ```suggestion
switch to a dedicated control mixin placed directly on the control element:
```

```html
<div class="esl-carousel-nav-container">
  <button esl-carousel-autoplay-control="pause">Pause autoplay</button>
  <esl-carousel esl-carousel-autoplay="8s">...</esl-carousel>
</div>
```

In other words:
- remove `control`, `controlBehaviour`, and `controlCls` from autoplay config
- register `ESLCarouselAutoplayControlMixin` where autoplay controls are needed
- move control behaviour to the control host via `esl-carousel-autoplay-control`

<a name="autoplay-control-mixin"></a>

## ESL Carousel Autoplay State Attribute

`esl-carousel-autoplay-state` is a read-only state mixin hosted directly on any element.

> It resolves a target carousel, listens for autoplay invalidation and reflects current autoplay state on the host.

Reviewer: ```suggestion
It resolves a target carousel, listens for autoplay state changes and reflects current autoplay state on the host.
```

</div>
```

### Runtime State Attributes

When attached to an element the mixin manages the following state markers on the host:

- `disabled` – autoplay plugin is not available on the target carousel
- `autoplay-state` – exclusive current state of the target autoplay instance. Possible values:
  - `unavailable` – no autoplay plugin found on the target carousel
  - `disabled` – autoplay exists but is currently disabled/stopped
  - `active` – autoplay cycle is running
  - `paused` – autoplay is paused explicitly by user action
  - `blocked` – autoplay is currently blocked by viewport / interaction / blocker conditions
  - `idle` – autoplay is enabled but there is no active cycle right now

`esl-carousel-autoplay-control` and `esl-carousel-autoplay-progress` extend the same state model and add their own behaviour on top.

## ESL Carousel Autoplay Control Attribute

`esl-carousel-autoplay-control` is a separate control mixin hosted directly on the control element.
It extends `esl-carousel-autoplay-state` and adds action/a11y semantics.

### Configuration

- `esl-carousel-autoplay-control` (`behaviour`) – primary attribute. Supported values:

>   - `stop` (default) – stop/start autoplay from scratch

Reviewer: ```suggestion
  - `stop` (default) – toggle autoplay, resetting the cycle on each start
```

>   - `pause` – pause/resume preserving current remaining time

Reviewer: ```suggestion
  - `pause` – toggle autoplay, resuming the cycle from the same point
```
